### PR TITLE
rbac: bug fixes on create sp for rbac commands

### LINKED ISSUE #700: az ad sp create-for-rbac: --years not working
looks like a parsing issue.

```
az ad sp create-for-rbac --years 2
ERROR: Non-integer years and months are ambiguous and not currently supported.
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/azure/cli/main.py", line 55, in main
    cmd_result = APPLICATION.execute(args)
  File "/usr/local/lib/python3.5/site-packages/azure/cli/application.py", line 117, in execute
    result = expanded_arg.func(params)
  File "/usr/local/lib/python3.5/site-packages/azure/cli/commands/__init__.py", line 216, in _execute_command
    result = operation(client, **kwargs) if client else operation(**kwargs)
  File "/usr/local/lib/python3.5/site-packages/azure/cli/command_modules/role/custom.py", line 464, in create_service_principal_for_rbac
    end_date = start_date + relativedelta(years=years)
  File "/usr/local/lib/python3.5/site-packages/dateutil/relativedelta.py", line 127, in __init__
    raise ValueError("Non-integer years and months are "
ValueError: Non-integer years and months are ambiguous and not currently supported.
```

### LINKED ISSUE #703: ad sp create: print client_id instead of SP URI/name?
cc @colemickens can weigh in.

I feel like most apps that run with Service Principal Accounts require inputs

- `client_id`
- `client_secret`

See the following output:

- it lists SP name, which is a URI (not clear if this can be used as `client_id` to the user)
    - there is an `--assignee` argument but not sure if that's objectID or SP ID.
- it lists the password (calls it “secret”), so I guess that one is fine

```
az ad sp create-for-rbac
WARNING: Service principal has been configured with name: 'http://azure-cli-2016-08-15-18-34-00', secret: '7dd33413-ade4-xxxx-xxxx-xxxxxxxxxxxx'
WARNING: Useful commands to manage azure:
WARNING: Assign a role:
WARNING:     az role assignment create --assignee b3c7a3d1-841c-4c80-b042-0c37a95c3423 --role Contributor
WARNING: Log in:
WARNING:     az login --service-principal -u http://azure-cli-2016-08-15-18-34-00 -p 7dd33413-ade4-xxxx-xxxx-xxxxxxxxxxxx --tenant xxxxxxx-xxxxx-xxxxx-xxxx-xxxxxx
WARNING: Reset credentials:
WARNING:     az ad sp reset-sp-credentials --name http://azure-cli-2016-08-15-18-34-00
WARNING: Revoke:
WARNING:     az ad sp delete --id http://azure-cli-2016-08-15-18-34-00
WARNING:     az ad app delete --id http://azure-cli-2016-08-15-18-34-00
```

### PR DESCRIPTION
**Pull request by yugangw-msft**
fix #700, fix #703
1. Ensure to parse `years` parameter as integer
2. Avoid referencing guid in the output of `create-for-rbac`

### CODE REVIEW

File: src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py [hunk 1]
```diff
@@ -511,24 +510,24 @@ def reset_service_principal_credential(name, secret=None, years=1):
...
-    logger.warning('    az role assignment create --assignee %s --role Contributor', sp_object_id)
+    logger.warning('    az role assignment create --assignee %s --role Contributor', sp_name)
     logger.warning('Log in:')
     logger.warning('    az login --service-principal -u %s -p %s --tenant %s',
                    sp_name, secret, tenant)
     logger.warning('Reset credentials:')
-    logger.warning('    az ad sp reset-sp-credentials --name %s', sp_name)
+    logger.warning('    az ad sp reset-credentials --name %s', sp_name)
     logger.warning('Revoke:')
-    logger.warning('    az ad sp delete --id %s', sp_name)
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py]: it is by design that in the developer tenant (which is our case), that deleting the application will cause service principal object to be deleted automatically. So we don't need this.

**Comment by derekbekoe** [comment]
:shipit:

**Comment by derekbekoe** [comment]
I modified the PR description to say "fix 700, fix 703" so that it references both issues. (https://help.github.com/articles/closing-issues-via-commit-messages/#closing-multiple-issues)

**Comment by tjprescott** [comment]
:shipit:

**Pull request merged**